helix-term/src/commands.rs

```diff
@@ -1030,20 +1030,29 @@ fn goto_file_impl(cx: &mut Context, action: Action) {
         .map(|r| text.slice(r.from()..r.to()).to_string())
         .collect();
     let primary = selections.primary();
-    if selections.len() == 1 && primary.to() - primary.from() == 1 {
-        let current_word = movement::move_next_long_word_start(
-            text.slice(..),
-            movement::move_prev_long_word_start(text.slice(..), primary, 1),
-            1,
-        );
+    if primary.len() == 1 {
+        let count = cx.count();
+        let current_word = selections
+            .clone()
+            .transform(|range| {
+                textobject::textobject_word(
+                    text.slice(..),
+                    range,
+                    textobject::TextObject::Inside,
+                    count,
+                    true,
+                )
+            })
+            .primary();
```

Comment:
You can operate on the `primary` range directly here:

```suggestion
        let current_word = textobject::textobject_word(
            text.slice(..),
            primary,
            textobject::TextObject::Inside,
            count,
            true,
        );
```